File: snippets/app/src/gms/java/com/google/maps/example/MapId.java
package com.google.maps.example;

import android.content.Context;
import com.google.android.gms.maps.GoogleMapOptions;
import com.google.android.gms.maps.MapView;
import com.google.android.gms.maps.SupportMapFragment;

class MapId {

  private void fragment() {
    // [START maps_android_support_map_fragment_map_id]
    GoogleMapOptions options = new GoogleMapOptions()
        .mapId("YOUR_MAP_ID");
    SupportMapFragment mapFragment = SupportMapFragment.newInstance(options);
    // [END maps_android_support_map_fragment_map_id]
  }

  private void mapView(Context context) {
    // [START maps_android_mapview_map_id]
    GoogleMapOptions options = new GoogleMapOptions()
        .mapId("YOUR_MAP_ID");
    MapView mapView = new MapView(context, options);
    // [END maps_android_mapview_map_id]
  }
}

File: snippets/app/src/gms/java/com/google/maps/example/kotlin/MapId.kt
package com.google.maps.example.kotlin

import android.content.Context
import com.google.android.gms.maps.GoogleMapOptions
import com.google.android.gms.maps.MapView
import com.google.android.gms.maps.SupportMapFragment

internal class MapId {
  private fun fragment() {
    // [START maps_android_support_map_fragment_map_id]
    val options = GoogleMapOptions()
      .mapId("YOUR_MAP_ID")
    val mapFragment = SupportMapFragment.newInstance(options)
    // [END maps_android_support_map_fragment_map_id]
  }

  private fun mapView(context: Context) {
    // [START maps_android_mapview_map_id]
    val options = GoogleMapOptions()
        .mapId("YOUR_MAP_ID")
    val mapView = MapView(context, options)
    // [END maps_android_mapview_map_id]
  }
}